lecture_parser.py

```diff
@@ -0,0 +1,82 @@
+from pptx import Presentation, slide
+
+
+class LectureParser:
+    def __init__(self, lecture_path: str):
+        """
+        Initializes a LectureParser object with the path to the lecture file.
+
+        :param lecture_path: The path to the lecture file.
+        """
+        self._lecture = lecture_path
+
+    def get_lecture_parts(self):
+        """
+        Retrieves the lecture parts.
+
+        This method is meant to be overridden by subclasses.
+
+        :return: An iterable of lecture parts.
+        """
+        raise NotImplementedError
+
+    def parse_lecture_part(self, lecture_part) -> str:
+        """
+        Parses a lecture part and returns the extracted text.
+
+        This method is meant to be overridden by subclasses.
+
+        :param lecture_part: The lecture part to parse.
+        :return: The extracted text from the lecture part.
+        """
+        raise NotImplementedError
+
+
+class PresentationParser(LectureParser):
+    def __init__(self, lecture_path):
+        """
+        Initializes a PresentationParser object with the path to the PowerPoint presentation.
+
+        :param lecture_path: The path to the PowerPoint presentation.
+        """
+        super().__init__(lecture_path)
+        self._parser = Presentation(self._lecture)
+
+    def get_lecture_parts(self):
+        """
+        Retrieves the slides from the PowerPoint presentation.
+
+        :return: An iterator of slides in the presentation.
+        """
+        for pres_slide in self._parser.slides:
+            yield pres_slide
+
+    def parse_lecture_part(self, pres_slide: slide.Slide) -> str:
+        """
+        Parses a slide from the PowerPoint presentation and returns the extracted text.
+
+        :param pres_slide: The slide to parse.
+        :return: The extracted text from the slide.
+        """
+        text_runs = []
+        for shape in pres_slide.shapes:
+            if not shape.has_text_frame:
+                continue
+            for paragraph in shape.text_frame.paragraphs:
+                for run in paragraph.runs:
+                    text_runs.append(run.text)

```

Comment:
Make this bit of code a little bit cleaner, see how you can use comprehension in some parts of the loop.
Replace for loops -  
https://medium.com/python-pandemonium/never-write-for-loops-again-91a5a4c84baf
Replace for loops with more advanced python features 

Replace these two for loops with a list comprehention with two iterators - 
https://stackoverflow.com/questions/18551458/how-to-frame-two-for-loops-in-list-comprehension-python

Reply:
done